internal/evaluation_target/application_snapshot_image/application_snapshot_image.go

```diff
@@ -199,11 +198,12 @@ func (a *ApplicationSnapshotImage) ValidateAttestationSignature(ctx context.Cont
 	// Extract the signatures from the attestations here in order to also validate that
 	// the signatures do exist in the expected format.
 	for _, att := range attestations {
-		signatures, err := entitySignatureFromAttestation(att)
+		sp, err := attestation.SLSAProvenanceFromLayer(att)
```

Comment:
A further step would be to make it so `SLSAProvenanceFromLayer` is only called once per attestation for a particular image. Maybe by stuffing the results somewhere in the `ApplicationSnapshotImage` object?

This could be done at a later time if execution time needs to be optimized.

Reply:
Thanks for going the extra mile to incorporate this! :pray: